src/main/test/at/ac/fhcampuswien/fhmdb/HomeControllerTest.java

```diff
@@ -1,7 +1,59 @@
 package at.ac.fhcampuswien.fhmdb;
 
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
 import static org.junit.jupiter.api.Assertions.*;
 
 class HomeControllerTest {
 
+    @Nested
+    public class GenreIsSelected {
+        @Test
+        public void whenGenreIsSelected_thenFilteredListShouldContainOnlyMoviesWithThatGenre() {
+            fail("Not implemented yet");
+        }
+    }
+
+    @Nested
+    public class SearchFieldIsFilled {
+        @Test
+        public void whenSearchFieldIsFilled_thenFilteredListShouldContainOnlyMoviesThatMatchSearchCriteria() {
+            fail("Not implemented yet");
+        }
+    }
+
+    @Nested
+    public class GenreIsSelectedAndSearchFieldIsFilled {
+        @Test
+        public void filteredListShouldContainOnlyMoviesThatMatchSearchCriteriaAndGenre() {
+            fail("Not implemented yet");
+        }
+
+        @Test
+        public void whenGenreIsReset_thenFilteredListShouldContainOnlyMoviesThatMatchSearchCriteria() {
+            fail("Not implemented yet");
+        }
+
+        @Test
+        public void whenSearchFieldIsReset_thenFilteredListShouldContainOnlyMoviesThatMatchGenre() {
+            fail("Not implemented yet");
+        }
+
+        @Test
+        public void SearchFieldIsResetAndThenGenreIsReset_thenFilteredListShouldContainAllMovies() {
+            fail("Not implemented yet");
+        }
+
+        @Test
+        public void SearchFieldIsResetAndThenGenreIsResetAndThenGenreIsSelectedAgain_thenFilteredListShouldContainOnlyMoviesThatMatchGenre() {
+            fail("Not implemented yet");
+        }
+
+        @Test
+        public void SearchFieldIsResetAndThenGenreIsResetAndThenSearchFieldIsFilledAgain_thenFilteredListShouldContainOnlyMoviesThatMatchSearchCriteria() {
+            fail("Not implemented yet");
+        }
+    }
```

Comment:
I think you can remove those test cases because it tests the same thing as the test that filters one criteria and the test which resets the filters.

If the selection of both works and the reset works, this test cases will also work. Therefore this would test the functionality a second time, but in one test.

Reply:
removed the test stubs